### PR TITLE
feat: Add Divider Component with Tests

### PR DESCRIPTION
**Pull request by noahwaldner**
-  Added Divider Component from Fondue
- Rewritten tests to vitest and teasting-library

### CODE REVIEW

File: packages/components/package.json [hunk 1]
```diff
@@ -44,6 +45,7 @@
         "eslint": "^8.56.0",
         "eslint-plugin-notice": "^0.9.10",
         "happy-dom": "^13.3.8",
+        "jest": "^29.7.0",
```
Comment (SamuelAlev) [inline packages/components/package.json:48]: Not necessary please remove (same for the types)

File: packages/components/src/components/Divider/Divider.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                    style={{
+                        marginLeft: parseInt(height) / 2,
+                        marginRight: parseInt(height) / 2,
+                    }}
+                >
+                    <div
+                        className={verticalClassNames}
+                        style={{ borderRightColor: color }}
+                        data-test-id="fondue-divider-line"
+                    ></div>
```
Comment (SamuelAlev) [inline packages/components/src/components/Divider/Divider.tsx:59]:
```suggestion
                    />
```

File: packages/components/src/components/Divider/Divider.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    vertical = false,
+    style = DividerStyle.Solid,
+    height = DividerHeight.Small,
+    'data-test-id': dataTestId = DIVIDER_TEST_ID,
+    color = '#CCC',
+}: DividerProps): ReactElement => {
+    const verticalClassNames = `tw-w-px tw-h-full tw-border-r tw-m-0 ${styleMap[style]}`;
+    return (
+        <>
+            {vertical && (
```
Comment (SamuelAlev) [inline packages/components/src/components/Divider/Divider.tsx:45]: make it a ternary?

File: packages/components/src/components/Divider/Divider.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    [DividerStyle.Solid]: 'tw-border-solid',
+    [DividerStyle.Dotted]: 'tw-border-dotted',
+};
+
+const DIVIDER_TEST_ID = 'fondue-divider';
+
+export const Divider = ({
+    vertical = false,
+    style = DividerStyle.Solid,
+    height = DividerHeight.Small,
```
Comment (SamuelAlev) [inline packages/components/src/components/Divider/Divider.tsx:38]: Please remove the enum and use union type (`'none' | 'dashed' | 'solid' | 'dotted'`)

File: packages/components/src/components/Divider/__tests__/divider.test.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+import { Divider, DividerHeight, DividerStyle } from '../Divider';
+
+const DEFAULT_COLOR_HEX = '#CCC';
+const COLOR_HEX = '#4065AE';
+const DIVIDER_SELECTOR = 'fondue-divider';
+const DIVIDER_LINE_SELECTOR = 'fondue-divider-line';
+
+describe('Divider component', () => {
+    test('loads and displays divider', async () => {
```
Comment (SamuelAlev) [inline packages/components/src/components/Divider/__tests__/divider.test.tsx:14]: `it('should [...]`

File: packages/components/src/components/Divider/__tests__/divider.test.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        render(<Divider />);
+        const divider = await screen.findByTestId(DIVIDER_LINE_SELECTOR);
+        expect(divider).toHaveStyle({ borderTopColor: DEFAULT_COLOR_HEX });
+    });
```
Comment (SamuelAlev) [inline packages/components/src/components/Divider/__tests__/divider.test.tsx:18]: Please add line breaks

File: packages/components/src/components/Divider/__tests__/divider.test.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    test('has correct color', async () => {
+        render(<Divider color={COLOR_HEX} />);
+        const divider = await screen.findByTestId(DIVIDER_LINE_SELECTOR);
+        expect(divider).toHaveStyle({ borderTopColor: COLOR_HEX });
+    });
+    test('allows the height to be set to Small', async () => {
```
Comment (SamuelAlev) [inline packages/components/src/components/Divider/__tests__/divider.test.tsx:24]: `it('should allow [...]`
(same for the rest)

File: packages/components/src/components/Divider/__tests__/divider.test.tsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    test('allows the height to be set to Large', async () => {
+        render(<Divider height={DividerHeight.Large} />);
+        const divider = await screen.findByTestId(DIVIDER_SELECTOR);
+        expect(divider).toHaveStyle({ height: DividerHeight.Large });
+    });
+
+    test('allows the divider border style to be dashed', async () => {
+        render(<Divider style={DividerStyle.Dashed} />);
+        const divider = await screen.findByTestId(DIVIDER_LINE_SELECTOR);
+        expect(divider).toHaveClass('tw-border-dashed');
```
Comment (SamuelAlev) [inline packages/components/src/components/Divider/__tests__/divider.test.tsx:45]:
```suggestion
        expect(divider).toHaveStyle({ borderStyle: 'dashed' });
```
?

File: packages/components/src/components/Divider/__tests__/divider.test.tsx [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+        expect(divider).toHaveStyle({ borderTopColor: DEFAULT_COLOR_HEX });
+    });
+    test('has correct color', async () => {
+        render(<Divider color={COLOR_HEX} />);
+        const divider = await screen.findByTestId(DIVIDER_LINE_SELECTOR);
+        expect(divider).toHaveStyle({ borderTopColor: COLOR_HEX });
+    });
+    test('allows the height to be set to Small', async () => {
+        render(<Divider height={DividerHeight.Small} />);
+        const divider = await screen.findByTestId(DIVIDER_SELECTOR);
```
Comment (SamuelAlev) [inline packages/components/src/components/Divider/__tests__/divider.test.tsx:26]:
```suggestion
        const { getByTestId } = render(<Divider height={DividerHeight.Small} />);
        const divider = getByTestId(DIVIDER_SELECTOR);
```
same everywhere

File: packages/components/src/components/Divider/__tests__/divider.test.tsx [hunk 10]
```diff
@@ -0,0 +1,77 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { render, screen } from '@testing-library/react';
```
Comment (SamuelAlev) [inline packages/components/src/components/Divider/__tests__/divider.test.tsx:3]:
```suggestion
import { render } from '@testing-library/react';
```

File: packages/components/src/components/Divider/__tests__/divider.test.tsx (hunk 8 above)
Comment (noahwaldner) [reply packages/components/src/components/Divider/__tests__/divider.test.tsx:51]: tailwind is not evaluated in the tests, thus, only the classes are present

**Pull request merged**